DIPs/DIP1xxx-Destructor.md

```diff
@@ -0,0 +1,60 @@
+# Destructor Tools
+
+| Field           | Value                                                           |
+|-----------------|-----------------------------------------------------------------|
+| DIP:            | (number/id -- assigned by DIP Manager)                          |
+| Review Count:   | 0 (edited by DIP Manager)                                       |
+| Author:         | Alexander Heistermann (alexanderheisterman@gmail.com)           |
+| Implementation: | None                                                            |
+| Status:         | Draft                                                           |
+
+## Abstract
+Opt-in static checking for destroy function and weaker destroy function named destructor_hook for classes.
+
+## Contents
+* [Rationale](#rationale)
+* [Description](#description)
+* [Reviews](#reviews)
+
+
+## Rational 
+
+The current �official� way of calling the D class deconstrutor is to call the destroy function.
```

Comment:
`destroy` is also code, so should be surrounded with back ticks.  Also, I think it would be worthwhile to explain why `destroy` can't be called in the context of `@safe` and `@nogc`.  Does the implementation of `destroy` just need to be changed, or is there a fundamental design flaw in the language?  If the latter, what is it?

Reply:
AFAIK Implementation. rt_finalize doesn't know anything about the object being passed as it coverts to void and this is an issue as c++ object can be allocated with the GC.